pkg/kubelet/dockershim/docker_container.go

```diff
@@ -307,7 +307,8 @@ func (ds *dockerService) StartContainer(_ context.Context, r *runtimeapi.StartCo
 // StopContainer stops a running container with a grace period (i.e., timeout).
 func (ds *dockerService) StopContainer(_ context.Context, r *runtimeapi.StopContainerRequest) (*runtimeapi.StopContainerResponse, error) {
 	err := ds.client.StopContainer(r.ContainerId, time.Duration(r.Timeout)*time.Second)
-	if err != nil {
+	// Do not return error if the container does not exist
+	if err != nil && !libdocker.IsContainerNotFoundError(err) {
```

Comment:
If the container wasn't stopped, I wouldn't expect to return a successful StopContainerResponse... I feel like this should be handled elsewhere.

Reply:
The func `IsContainerNotFoundError` also used in `StopPodSandbox` and `RemovePodSandbox`.
https://github.com/kubernetes/kubernetes/blob/fbffe056dd03bd9c746e8819ad22043d640a4489/pkg/kubelet/dockershim/docker_sandbox.go#L240
https://github.com/kubernetes/kubernetes/blob/fbffe056dd03bd9c746e8819ad22043d640a4489/pkg/kubelet/dockershim/docker_sandbox.go#L274